kotlin-runtime/ftl-runtime/src/test/kotlin/xyz/block/ftl/schemaextractor/ExtractSchemaRuleTest.kt

```diff
@@ -324,33 +322,34 @@ internal class ExtractSchemaRuleTest(private val env: KotlinCoreEnvironment) {
   @Test
   fun `fails if http ingress without http request-response types`() {
     val code = """
-        package ftl.echo
+ /**
+ * Echo module.
+ */
+@file:Module
```

Comment:
Do we even need this annotation anymore do you think?

Reply:
are you thinking instead we'd just detect packages beginning with `ftl.`? i like the ease of that but in a way it's also nice to have the annotation explicitly denote the API layer for a module

could see a world where new developers onboarding to FTL accidentally introduce schema-breaking changes by adding new arbitrary packages prefixed with `ftl.`, or unexpected things make it into the module schema etc. i'm thinking the less people have to know about the "rules" of FTL the better, and having the annotation means more things are by default ignored rather than by default opted-in

what are your thoughts?

Reply:
> are you thinking instead we'd just detect packages beginning with ftl.?

Yeah that was my thinking. We already require this anyway, it seems redundant to also have an annotation. I've been pondering the same for Go.

> could see a world where new developers onboarding to FTL accidentally introduce schema-breaking changes by adding new arbitrary packages prefixed with ftl.

Can you elaborate on how this would happen? The "roots" of the FTL schema are the verbs, so there's no real way to accidentally introduce a breaking change implicitly. Users always have to explicitly tag functions with `@Verb`.

Reply:
ah ok - in Kotlin the root is no longer verb, at least with the changes in this PR. instead it would be the module annotation which would allow users to define verb-less modules.

re: schema-breaking confusion; i was thinking of a case where someone makes a nested package like `ftl.foo.bar`, where `ftl.foo` is their module package. with the module annotation, this would be ignored because we're initiating extraction from annotated files. if instead we looked for `ftl.` to extract schema, this would break on the grounds of being an improper module package name where the required structure is `ftl.<module>`. this feels like it would be confusing to someone whose intention isn't to introduce a new module, just add a nested package

this is only an issue if we extract from the module (package) as the "root" rather than verb annotations - but maybe there's a bigger question here of whether we want to support verb-less modules at all?

Reply:
Ah I see where the confusion lies, I wasn't aware that you were making that change. Let's revert that change for now as it needs a bit more thought, and we'd also need an equivalent solution for Go. I'm leaning towards explicitly annotating types FWIW, because I think it is more explicit than exporting everything from a module. Particularly in Go, where an annotation on a single file actually ends up being applied to the whole package.